### ISSUE TITLE
Autoscroll

### ISSUE THREAD

**Issue by moatn** [joedf/Qonsole#1]
Hey Joedf,

Qonsole works very great, though i wanted to ask if its normal behavior that the autoscroll function doesnt always works. When multiple lines go through the Qonsole i have to scroll manual down, is there any way to fix this?

Keep up the great work!

**Comment by joedf** [comment]
Hey there, are you using mintty or cygwin by any chance?
_Possibly related:_ https://github.com/joedf/Qonsole/issues/2

**Comment by moatn** [comment]
Nope, ive redirected the terminal to a Powershell shortcut so i have admin rights. This way i can use the Qonsole with full rights and as packet manager etc.

**Comment by joedf** [comment]
Ahhh okkay
Could you please post your Qonsole configuration file (INI)? :

---

**Comment by joedf** [comment]
@MaartenPentinga Hi there?
Can you please post your Config file?

**Comment by moatn** [comment]
Hey Joe!

Sorry for the slow reaction :)

``` INI
[Animation]
Speed=10
Delay=10
dx=25
AnimationDisabled=0
GuiBGDarken_Increment=6
GuiBGDarken_Max=127
GuiBGDarken_Color=0x1A1A1A
[Settings]
HideOnInActive=0
HorizontallyCentered=0
BottomPlaced=0
CmdPaste=1
AutoWinActivate=0
ReduceMemory=1
RunOnStartUp=1
Console_Mode=Cmd
TransparencyPercent=10
CMD_Path=C:\Users\Nerf\Documents\Quake\powershell_Qonsole.lnk
OpenHotkey=#`
CMD_Width=1374
CMD_Height=400
CMD_StartUpArgs=
```

**Comment by joedf** [comment]
@MaartenPentinga 
Try this test version.
[qonsole-[143b_a1].zip](https://github.com/joedf/Qonsole/files/190513/qonsole-.143b_a1.zip)

**Comment by moatn** [comment]
@joedf It seems to work now. Though i did some testing. It seems if i make the height to big it stops autoscrolling. 

So my quess it has to do with the height and not the terminal that i am emuluting within your program.

**Comment by joedf** [comment]
Thank you. I will soon ask you to try out another test version, once I have one.

**Comment by joedf** [comment]
@MaartenPentinga Sorry about the late reply. If you press the [END] key, does it scroll to your line?
Also if you manually scroll, does it display properly or does it leave a black area under the prompt line?

**Comment by TrekkieGod** [comment]
I have the same problem, and the one additional piece of information I can offer is that I think this issue is unique to Windows 10.  I've installed it on four different machines.  Two of them are Windows 10 machines, and they both have the autoscroll problem.  The other two are Windows 8.1, and there are no problems whatsoever.

**Comment by TrekkieGod** [comment]
I discovered that going to Qonsole Settings / Console Settings / Options tab and checking the "Use legacy console" box fixes the autoscroll issues for me. So I think the problem might be specific with the new Windows Console present in Windows 10 and Windows Server 2016.

**Comment by joedf** [comment]
I think you're right. Thanks, I'll have to add a notice and maybe have something automatically change it.

**Comment by joedf** [comment]
More info about it here: https://technet.microsoft.com/en-us/library/mt427362.aspx
It must be because of the new line wrap and resize features.

**Comment by joedf** [comment]
I have found a workaround for this. I've noticed that simply sending a mouse wheeldown seems to work fine. I have a few other things to test and fix too (will be in v1.4.5). I should have this specific bug patched in v1.4.4 which should be up by tomorrow (using the workaround).

**Comment by joedf** [comment]
Won't be released finally, The problem is a little more complex than expected. The console buffer size is not resizing correctly so when the window is cropped to a smaller size, the "bug" happens...
